include/grpcpp/support/proto_buffer_reader.h

```diff
@@ -189,10 +189,16 @@ class ProtoBufferReader : public grpc::protobuf::io::ZeroCopyInputStream {
   // This function takes ownership of slice and return a newly created Cord off
   // of it.
   static absl::Cord MakeCordFromSlice(grpc_slice slice) {
+    grpc_slice* slice_for_cord = new grpc_slice;
```

Comment:
Please add a comment describing why a slice may need to be on the heap.

It might also be valuable to have a `GRPC_SLICE_UNDO_INLINING` macro, which could avoid this juggling.

Reply:
I added a comment explaining why. Thanks!